hail/src/test/scala/is/hail/types/virtual/TStructSuite.scala

```diff
@@ -0,0 +1,44 @@
+package is.hail.types.virtual
+
+import is.hail.HailSuite
+import org.testng.Assert.{assertFalse, assertTrue}
+import org.testng.annotations.{DataProvider, Test}
+
+class TStructSuite extends HailSuite {
+  @DataProvider(name = "isPrefixOf")
+  def isPrefixOfData: Array[Array[Any]] =
+    Array(
+      Array(TStruct.empty, TStruct.empty, true),
+      Array(TStruct.empty, TStruct("a" -> TVoid), true),
+      Array(TStruct("a" -> TVoid), TStruct.empty, false),
+      Array(TStruct("a" -> TVoid), TStruct("a" -> TVoid), true),
+      Array(TStruct("a" -> TVoid), TStruct("a" -> TVoid, "b" -> TVoid), true),
+      Array(TStruct("a" -> TVoid, "b" -> TVoid), TStruct("a" -> TVoid), false),
+      // isPrefixOf ignores field names and compares the ordered sequence of types.
+      // Consider joins for example - we only care that the key fields have the same types
+      // so we compare the key types (which are structs) for equality ignoring field names.
+      // isPrefixOf is used in similar cases involving key types where we don't care about names.
+      Array(TStruct("b" -> TVoid), TStruct("a" -> TVoid, "b" -> TVoid), true)
+    )
+
+  @Test(dataProvider = "isPrefixOf")
+  def testIsPrefixOf(a: TStruct, b: TStruct, isPrefix: Boolean): Unit =
+    assert(a.isPrefixOf(b) == isPrefix, s"expected $a `isPrefixOf` $b == $isPrefix")
```

Comment:
I wasn't sure why we ignore field names but after Patrick's explaination (comment is lifted from a DM with him), I wanted to write this. Though trivial, the intent is to lock down behaviour for posterity that's decoupled from other logic.

Reply:
that's weird behavior, we could probably rewrite all usages of isPrefixOf to do something clearer. Good to add a test, though.